packages/discord.js/src/structures/MessageMentions.js

```diff
@@ -186,10 +199,12 @@ class MessageMentions {
     if (this._channels) return this._channels;
     this._channels = new Collection();
     let matches;
-    while ((matches = this.constructor.ChannelsPattern.exec(this._content)) !== null) {
-      const chan = this.client.channels.cache.get(matches[1]);
-      if (chan) this._channels.set(chan.id, chan);
+
+    while ((matches = this.constructor.GlobalChannelsPattern.exec(this._content)) !== null) {
+      const channel = this.client.channels.cache.get(matches.groups.id);
+      if (channel) this._channels.set(channel.id, channel);
     }
```

Comment:
Don't we have to reset lastIndex after?

Reply:
while loop will stop after matching all matches. So once lastIndex becomes greater than input length, it'll become zero again